app/controllers/users.js

```diff
@@ -4,20 +4,45 @@ const { validationError } = require('../errors'),
   { hashPassword, comparePasswords } = require('../helpers/hasher'),
   { createToken } = require('../helpers/token');
 
+const signUpWithAdminCondition = userData => {
+  const { firstName, lastName, email, password, isAdmin } = userData;
+  if (isAdmin) {
+    logger.info(`Creating user ${firstName} as admin`);
+    return hashPassword(password).then(hash => User.createUserAdmin(firstName, lastName, email, hash));
+  }
+  logger.info(`Creating user ${firstName}`);
+  return hashPassword(password).then(hash => User.createUser(firstName, lastName, email, hash));
```

Comment:
That is what I tried to do first, but Linter threw this error:
`Arrow function has too many parameters (5). Maximum allowed is 4.eslint(max-params)`.
So the two options were receiving an object as parameter, or have that two functions. I will change it to receive an object, and having one function.

Reply:
Even if linter is crying about max params, you can send the rest of the params as an object for example. So the function could be `User.createUser = (userParams, isAdmin = false)`. However, in this case it would force you to change how the fucntion is used which is a bit annoying.
I'm not too keen on that max params rule myself. You can ignore this one for example by doing 

\/* eslint-disable max-params \*/
Rule violating code here
\/* eslint-enable max-params \*/

Be very careful with this however and only do it when you have a justifiable reason to do so.